nixos/modules/services/networking/jitsi-meet.nix

```diff
@@ -0,0 +1,398 @@
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_NAME" = "videobridge";
+    "-Djava.util.logging.config.file" = "/etc/jitsi/videobridge/logging.properties";
+  };
+  jvbConfig = attrsToCfg cfg.videobridge.config;
+
+  jicofoArgs = {
+    "--host" = "localhost";
+    "--domain" = cfg.hostName;
+    "--secret" = "\${JICOFO_COMPONENT_SECRET}";
+    "--user_domain" = "auth.${cfg.hostName}";
+    "--user_name" = "focus";
+    "--user_password" = "\${JICOFO_USER_SECRET}";
+  };
+  jicofoProps = {
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_LOCATION" = "/etc/jitsi";
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_NAME" = "jicofo";
+    "-Djava.util.logging.config.file" = "/etc/jitsi/jicofo/logging.properties";
+    "-Djavax.net.ssl.trustStore" = "${dataDir}/prosody-ca.jks";
+    "-Djavax.net.ssl.trustStorePassword" = "javapls";
+  };
+  jicofoConfig = attrsToCfg cfg.jicofo.config;
+
+  defaultCfg = {
+    hosts = {
+      domain = cfg.hostName;
+      muc = "conference.${cfg.hostName}";
+      bridge = "jitsi-videobridge.${cfg.hostName}";
+    };
+    useNicks = false;
+    bosh = "//${cfg.hostName}/http-bind";
+    enableWelcomePage = true;
+  };
+
+  # default logging.properties modified to work better with journald - no timestamp or progname
+  loggingProperties = ''
+    handlers= java.util.logging.ConsoleHandler
+    java.util.logging.ConsoleHandler.level = ALL
+    java.util.logging.ConsoleHandler.formatter = net.java.sip.communicator.util.ScLogFormatter
+    .level=INFO
+    net.sf.level=SEVERE
+    net.java.sip.communicator.plugin.reconnectplugin.level=FINE
+    org.ice4j.level=SEVERE
+    org.jitsi.impl.neomedia.level=SEVERE
+    net.java.sip.communicator.service.resources.AbstractResourcesService.level=SEVERE
+    net.java.sip.communicator.util.ScLogFormatter.disableTimestamp=true
+  '';
+
+in
+{
+  options.services.jitsi-meet = with types; {
+    enable = mkEnableOption "Jitsi Meet - Secure, Simple and Scalable Video Conferences";
+
+    hostName = mkOption {
+      type = str;
+      example = "meet.example.org";
+      description = ''
+        Hostname of the Jitsi Meet instance.
+      '';
+    };
+
+    config = mkOption {
+      type = attrs;
+      default = {};
+      example = literalExample ''
+        {
+          enableWelcomePage = false;
+          defaultLang = "fi";
+        }
+      '';
+      description = ''
+        Contents of the <filename>config.js</filename> file containing the web app configuration.
+
+        See <link xlink:href="https://github.com/jitsi/jitsi-meet/blob/master/config.js" /> for an
+        example configuration with comments.
+      '';
+    };
+
+    videobridge = {
+      config = mkOption {
+        type = attrsOf str;
+        default = {
+          "org.jitsi.videobridge.AUTHORIZED_SOURCE_REGEXP" = "focus@auth.${cfg.hostName}/.*";
+          "org.jitsi.videobridge.TCP_HARVESTER_PORT" = "4443";
+          "org.jitsi.videobridge.SINGLE_PORT_HARVESTER_PORT" = "10000";
+        };
```

Comment:
What's the best way to extend this? Should we add an `extraConfig` option here so it's not necessary to override the whole default?

Reply:
Good point, I've changed the module so that it's not necessary to override the whole default now. Having two similar options would be confusing IMO.

Also changed the semantics of `services.jitisi-meet.config` so it's now deep-merged with defaults, before it was only possible to override top-level keys.